launcher/src/main/java/org/apache/spark/launcher/SparkSubmitCommandBuilder.java

```diff
@@ -229,7 +238,7 @@
       args.add(join(",", pyFiles));
     }
 
-    if (isExample) {
+    if (isExample && requiresMainClass) {
```

Comment:
In what situation will `requiresAppResource` be different from `requiresMainClass` at this point?

It seems to me they're basically the same. The only point in your code where they differ is in the no-arg constructor, which is not used by the `run-example` script.

Seems to me like you could rename the existing variable to `isSpecialCommand` or something and use it instead for both cases.

Reply:
Originally I was hesitating to merge them and loose the 2 different meaning but it's better. Added a small comment.